File: src/test/java/akostenko/aicars/math/PolarTest.java
package akostenko.aicars.math;

import static akostenko.aicars.math.Vector.PRECISION;
import static java.lang.StrictMath.PI;
import static java.lang.StrictMath.sqrt;
import static org.junit.Assert.*;

import org.junit.Test;

public class PolarTest {

    @Test
    public void dot_orthogonal() throws Exception {
        Vector a = new Decart(1, 0);
        Vector b = new Decart(0, 1);
        double result = a.dot(b);
        assertEquals(0, result, PRECISION);
    }

    @Test
    public void dot_collinear() throws Exception {
        Vector a = new Decart(2, 0);
        Vector b = new Decart(-3, 0);
        double result = a.dot(b);
        assertEquals(-6, result, PRECISION);
    }

    @Test
    public void dot_commutative() throws Exception {
        Vector a = new Polar(2, PI+PI/4);
        Vector b = new Polar(3, PI+PI/2);
        double result1 = a.dot(b);
        double result2 = b.dot(a);

        assertEquals(3*sqrt(2), result1, PRECISION);
        assertEquals(3*sqrt(2), result2, PRECISION);
    }

    @Test
    public void cross_orthogonal() throws Exception {
        Vector a = new Decart(2, 0);
        Vector b = new Decart(0, -3);
        double result = a.cross(b);
        assertEquals(-6, result, PRECISION);
    }

    @Test
    public void cross_collinear() throws Exception {
        Vector a = new Decart(2, 0);
        Vector b = new Decart(-3, 0);
        double result = a.cross(b);
        assertEquals(0, result, PRECISION);
    }

    @Test
    public void cross_antiCommutative() throws Exception {
        Vector a = new Polar(2, PI+PI/4);
        Vector b = new Polar(3, PI+PI/2);
        double result1 = a.cross(b);
        double result2 = b.cross(a);

        assertEquals(result1, -result2, PRECISION);
    }

}

File: src/test/java/akostenko/aicars/math/PolarTest.kt
package akostenko.aicars.math

import akostenko.aicars.math.Vector.PRECISION
import java.lang.StrictMath.PI
import java.lang.StrictMath.sqrt
import org.junit.Assert.*

import org.junit.Test

class PolarTest {

    @Test
    @Throws(Exception::class)
    fun dot_orthogonal() {
        val a = Decart(1.0, 0.0)
        val b = Decart(0.0, 1.0)
        val result = a.dot(b)
        assertEquals(0, result, Companion.getPRECISION())
    }

    @Test
    @Throws(Exception::class)
    fun dot_collinear() {
        val a = Decart(2.0, 0.0)
        val b = Decart(-3.0, 0.0)
        val result = a.dot(b)
        assertEquals(-6, result, Companion.getPRECISION())
    }

    @Test
    @Throws(Exception::class)
    fun dot_commutative() {
        val a = Polar(2.0, PI + PI / 4)
        val b = Polar(3.0, PI + PI / 2)
        val result1 = a.dot(b)
        val result2 = b.dot(a)

        assertEquals(3 * sqrt(2.0), result1, Companion.getPRECISION())
        assertEquals(3 * sqrt(2.0), result2, Companion.getPRECISION())
    }

    @Test
    @Throws(Exception::class)
    fun cross_orthogonal() {
        val a = Decart(2.0, 0.0)
        val b = Decart(0.0, -3.0)
        val result = a.cross(b)
        assertEquals(-6, result, Companion.getPRECISION())
    }

    @Test
    @Throws(Exception::class)
    fun cross_collinear() {
        val a = Decart(2.0, 0.0)
        val b = Decart(-3.0, 0.0)
        val result = a.cross(b)
        assertEquals(0, result, Companion.getPRECISION())
    }

    @Test
    @Throws(Exception::class)
    fun cross_antiCommutative() {
        val a = Polar(2.0, PI + PI / 4)
        val b = Polar(3.0, PI + PI / 2)
        val result1 = a.cross(b)
        val result2 = b.cross(a)

        assertEquals(result1, -result2, Companion.getPRECISION())
    }

}